src/models/krate.rs

```diff
@@ -433,18 +432,32 @@ impl Crate {
         conn: &mut PgConnection,
         options: PaginationOptions,
     ) -> QueryResult<(Vec<ReverseDependency>, i64)> {
+        use diesel::dsl::{exists, not, sql};
         use diesel::sql_query;
         use diesel::sql_types::{BigInt, Integer};
 
         let offset = options.offset().unwrap_or_default();
-        let rows: Vec<WithCount<ReverseDependency>> =
+        let rows: Vec<ReverseDependency> =
             sql_query(include_str!("krate_reverse_dependencies.sql"))
                 .bind::<Integer, _>(self.id)
                 .bind::<BigInt, _>(offset)
                 .bind::<BigInt, _>(options.per_page)
                 .load(conn)?;
 
-        Ok(rows.records_and_total())
+        let dep_versions_subq = dependencies::table
+            .filter(dependencies::crate_id.eq(self.id))
+            .select(dependencies::version_id);
+        let yanked_subq = versions::table
+            .filter(versions::id.eq(default_versions::version_id))
+            .filter(versions::yanked)
+            .select(sql::<Integer>("1"));
+        let total = default_versions::table
+            .filter(default_versions::version_id.eq_any(dep_versions_subq))
+            .filter(not(exists(yanked_subq)))
+            .count()
+            .get_result(conn)?;
```

Comment:
is running two queries after another really faster than doing it in a single query? I guess it would give us the option of caching the count separately, but other than that I would expect this to be slower than the alternative 😅

Reply:
You're right, after running some local benchmarks, it seems the single query is still slightly faster (~98ms vs ~103ms). Based on this, I'll revert the change back to using a single query.